mpt_api_client/models/collection.py

```diff
@@ -1,54 +1,34 @@
 from collections.abc import Iterator
-from typing import Any, ClassVar, Self, override
 
-from httpx import Response
-
-from mpt_api_client.models.base import BaseCollection, ResourceData
 from mpt_api_client.models.meta import Meta
-from mpt_api_client.models.resource import Resource
+from mpt_api_client.models.model import Model, ResourceData
 
+ResourceList = list[ResourceData]
 
-class Collection[ResourceType](BaseCollection):
-    """Provides a base collection to interact with api collection data using fluent interfaces."""
 
-    _data_key: ClassVar[str] = "data"
-    _resource_model: type[Resource] = Resource
+class Collection[ItemType: Model]:
+    """Provides a collection to interact with api collection data using fluent interfaces."""
 
-    def __init__(
-        self, collection_data: list[ResourceData] | None = None, meta: Meta | None = None
-    ) -> None:
+    def __init__(self, items: list[ItemType] | None = None, meta: Meta | None = None) -> None:  # noqa: WPS110
```

Comment:
Why not change the items name instead of adding the noqa? 

The typing feels a bit strange since in the end this is a list of models, right? We should keep the naming consistent across the code